src/management/list.ts

```diff
@@ -34,6 +35,67 @@ export async function listFirebaseProjects(
   }
 }
 
+/**
+ * Send recurring API requests to list all Firebase apps of a Firebase project
+ * @return a promise that resolves to the list of all Firebase apps.
+ */
+export async function listFirebaseApps(
+  projectId: string,
+  platform?: AppPlatform,
+  pageSize: number = APP_LIST_PAGE_SIZE
+): Promise<AppMetadata[]> {
+  const apps: AppMetadata[] = [];
+  try {
+    let nextPageToken = "";
+    do {
+      const response = await getPageApiRequest(
+        getListAppsResourceString(projectId, platform),
+        pageSize,
+        nextPageToken
+      );
+      if (response.body.apps) {
+        const appsOnPage = response.body.apps.map(
+          (app: any) => (app.platform ? app : { ...app, platform })
```

Comment:
Can a comment be added here regarding why `app.platform` may not exist and what that means? It's a bit of a mystery why this map is necessary

Reply:
I'm using different endpoints for listing apps (see [rest api endpoints](https://docs.google.com/document/d/1jkwfbu-3H4MNmvKXloMl5SonYxodWuFenZaYLF2_tFo/edit#heading=h.8x9blrb9ezuu)). Platform is implied for the later 3

I will add a comment in code

Reply:
**Testing:** I didn't really test this behavior... I think I will add 3 more helper methods for generating app list for 3 supporting platforms, whose responses don't have `platform` field.

Reply:
Done